templates/generic/service_and_deployment.yaml

```
@@ -0,0 +1,156 @@
+{{- $geoserverName := (include "geoserver.fullname" .) -}}
+{{- $geoserverLabels := (include "geoserver.labels" .) -}}
+{{- $geoserverSelectorLabels := (include "geoserver.selectorLabels" .) -}}
+{{- $geoserverServiceAccountName := (include "geoserver.serviceAccountName" .) }}
+{{- $chart := .Chart -}}
+{{- $shaConfigmap := (include (print $.Template.BasePath "/config/configmap.yaml") . | sha256sum) -}}
+
+{{- $values := .Values -}}
+
+{{- range $serviceName, $serviceDefinition := $values.geoserver.services }}
+---
+# service {{ $serviceName }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ $geoserverName }}-{{ $serviceName }}
+  labels:
+    {{- $geoserverLabels | nindent 4 }}
+    gs-service: {{ $serviceName }}
+spec:
+  type: {{ $serviceDefinition.service.type }}
+  ports:
+    - port: {{ $serviceDefinition.service.port }}
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    {{ $geoserverSelectorLabels | nindent 4  }}
+    gs-service: {{ $serviceName }}
+
+---
+# deployment {{ $serviceName }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $geoserverName }}-{{ $serviceName }}
+  labels:
+    {{- $geoserverLabels  | nindent 4 }}
+    gs-service: rest
+  annotations:
+    argocd.argoproj.io/sync-wave: "9"
+spec:
+  revisionHistoryLimit: 3
+  replicas: {{ $serviceDefinition.replicaCount }}
+  selector:
+    matchLabels:
+      {{ $geoserverSelectorLabels | nindent 6 }}
+      gs-service: rest
+  template:
+    metadata:
+      annotations:
+      {{- with $serviceDefinition.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+        checksum/config: {{ $shaConfigmap }}
+      labels:
+        {{ $geoserverSelectorLabels | nindent 8 }}
+        gs-service: rest
+        {{- if $values.global.custom_pod_labels }}
+          {{- toYaml $values.global.custom_pod_labels | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ $geoserverServiceAccountName }}
+      securityContext:
+        {{- toYaml $serviceDefinition.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ $chart.Name }}-{{ $serviceName }}
```

Can simply be 
```suggestion
        - name: {{ $serviceName }}
```
Or
```suggestion
        - name: main
```